dart/lib/src/hub.dart

```diff
@@ -24,13 +26,18 @@ class Hub {
 
   final SentryOptions _options;
 
+  late SentryTracesSampler _tracesSampler;
+
+  final Map<dynamic, _Pair<ISentrySpan, String>> _throwableToSpan = {};
```

Comment:
since Dart does not have a weak reference, we have to rotate that

Reply:
an ide would be to use either https://pub.dev/packages/weak_map or https://api.flutter.dev/flutter/dart-core/Expando-class.html
could also work for the `DeduplicationEventProcessor`

@ueman have you used this already?

Reply:
@bruno-garcia opinions? that could hold and leak unwanted objects

Reply:
I haven't used `Expando`s yet. As far as I understand those aren't like a weak reference in Java. The `weak_map` itself says that the map only keeps keys with values as long as the user keeps the key. 

https://pub.dev/packages/weak_map#why-doesnt-dart-allow-for-real-weak-references-anyway

Reply:
yep, its somewhat similar

Reply:
`Expando` sounds like a viable solution